pkg/cmd/init.go

```diff
@@ -512,3 +518,30 @@ func certsToRegenerate(cs *certchains.CertificateChains) ([][]string, error) {
 
 	return regenCerts, err
 }
+
+func cleanupStaleKubeconfigs(cfg *config.Config) error {
+	currentKubeconfigs := make(map[string]struct{})
+	for _, name := range append(cfg.ApiServer.SubjectAltNames, cfg.Node.HostnameOverride) {
+		currentKubeconfigs[name] = struct{}{}
+	}
+	files, err := os.ReadDir(cfg.KubeConfigRootAdminPath())
+	if err != nil {
+		return err
+	}
+	deleteDirs := make([]string, 0)
+	for _, file := range files {
+		if !file.IsDir() {
+			continue
+		}
+		if _, ok := currentKubeconfigs[file.Name()]; !ok {
+			deleteDirs = append(deleteDirs, filepath.Join(cfg.KubeConfigRootAdminPath(), file.Name()))
+		}
+	}
+	klog.Infof("Stale kubeconfig directories: %v", deleteDirs)
```

Comment:
Maybe instead of logging a list here, the loop below could log right before it tries to remove each item? That would make the logs easier to read and would fit with the "unit of work" style logging. https://docs.openstack.org/oslo.log/latest/user/guidelines.html#overall-logging-principles

Reply:
Didnt know about this guide, it is done now.